### DISCUSSION TITLE
How to get `request_id` from request use `axum`?

### DISCUSSION THREAD

**Discussion by holicc** [aws/aws-lambda-rust-runtime#808] [Q&A] [answered]
handler function :
```rust
async fn report<C: ServerContext>(State(ctx): State<Arc<C>>) -> Result<String, Error> {
    ...
}
```

**Comment by calavera** [comment, marked as answer]
You need to extract it from the `Request` object. Here's an example: https://github.com/awslabs/aws-lambda-rust-runtime/blob/main/examples/http-axum-middleware/src/main.rs#L18-L24
